python/semantic_kernel/connectors/ai/open_ai/services/open_ai_handler.py

```diff
@@ -102,6 +107,72 @@ def store_usage(self, response):
             if hasattr(response.usage, "completion_tokens"):
                 self.completion_tokens += response.usage.completion_tokens
 
-    def get_request_settings_class(self) -> "AIRequestSettings":
-        """Return the class with the applicable request settings."""
-        return OpenAIRequestSettings
+    def _get_metadata_from_chat_response(self, response: ChatCompletion) -> Dict[str, Any]:
+        return {
+            "id": response.id,
+            "created": response.created,
+            "system_fingerprint": response.system_fingerprint,
+            "usage": response.usage,
+        }
+
+    def _get_metadata_from_text_response(self, response: Completion) -> Dict[str, Any]:
+        return {
+            "id": response.id,
+            "created": response.created,
+            "system_fingerprint": response.system_fingerprint,
+            "usage": response.usage,
+        }
+
+    def _get_metadata_from_streaming_chat_response(self, response: ChatCompletionChunk) -> Dict[str, Any]:
+        return {
+            "id": response.id,
+            "created": response.created,
+            "system_fingerprint": response.system_fingerprint,
+        }
+
+    def _get_metadata_from_streaming_text_response(self, response: Completion) -> Dict[str, Any]:
+        return {
+            "id": response.id,
+            "created": response.created,
+            "system_fingerprint": response.system_fingerprint,
+        }
+
+    def _get_metadata_from_chat_choice(self, choice: Union[Choice, ChunkChoice]) -> Dict[str, Any]:
+        return {
+            "finish_reason": choice.finish_reason,
+            "index": choice.index,
+            "logprobs": choice.logprobs,
+        }
+
+    def _get_tool_calls_from_chat_choice(self, choice: Union[Choice, ChunkChoice]) -> Optional[List[ToolCall]]:
+        if isinstance(choice, Choice):
+            content = choice.message
+        else:
+            content = choice.delta
+        if content.tool_calls is None:
+            return None
+        return [
+            ToolCall(
+                id=tool.id,
+                type=tool.type,
+                function=FunctionCall(name=tool.function.name, arguments=tool.function.arguments),
+            )
+            for tool in content.tool_calls
+            if tool.function is not None
+        ]
+
+    def _get_function_call_from_chat_choice(self, choice: Union[Choice, ChunkChoice]) -> Optional[FunctionCall]:
```

Comment:
Isn't `function_call` going away, or is this for backwards compatibility? When would we be able to remove this and only rely on tool calls?